llvm/tools/objwriter/objwriter.cpp

```diff
@@ -299,8 +310,11 @@ MCSection *ObjectWriter::GetSpecificSection(const char *SectionName,
     } else if (attributes & CustomSectionAttributes_Writeable) {
       Flags |= ELF::SHF_WRITE;
     }
+    unsigned SectionType = (attributes & CustomSectionAttributes_Uninitialized)
+                               ? ELF::SHT_NOBITS
```

Comment:
for BSS section the SHT_NOBITS should be combined with SHF_ALLOC, to be present in memory, which we get from line 301 above and SHF_WRITE, which we get because we set the writeable attribute in object writer at:

```  c
              case SectionType.Uninitialized:
                    attributes |= CustomSectionAttributes.Uninitialized | CustomSectionAttributes.Writeable;
                    break;
```

Looks right to me.